### PR TITLE
New deck workflow improvements

### LINKED ISSUE #1489: New deck workflow improvements
on the public tenant: `/decks/new/`

- [x] Set the owner's username to firstname.lastname (instead of "owner")
- [x] Set the owner's password to firstname-deckname-lastname
- [x] FIX broken email verification link that is sent (it does not use the subdomain, so the link doesn't work.  e.g. bytedeck.com/accounts/confirm-email/.... instead of deckname.bytedeck.com/account/confirm-email/...
- [ ] Once the owner user has verified the email (need to hook into that somehow?) for the first time, send an email with instructions for how to log in with the username and password.

### PR DESCRIPTION
**Pull request by bashu**
Please ensure you are familiar with our Pull Request Description expectations described here: https://www.pullrequest.com/blog/writing-a-great-pull-request-description/
### What?

This PR is for features/improvements requested in https://github.com/bytedeck/bytedeck/issues/1489, request was to made workflow improvements on the public tenant: `/decks/new/`

### Why?

It was requested in https://github.com/bytedeck/bytedeck/issues/1489

### How?

> Set the owner's username to firstname.lastname (instead of "owner")

``generate_unique_username`` (part of `allauth` app) method returns a unique username from the combination of strings present in txts (first argument) iterable, for reference:

https://docs.allauth.org/en/latest/account/advanced.html#creating-and-populating-user-instances

> Set the owner's password to firstname-deckname-lastname

Solved by explicitly calling `User.set_password` method and setting new password after creation form was validated successfuly, see `form_valid` method.

> Once the owner user has verified the email (need to hook into that somehow?) for the first time, send an email with instructions for how to log in with the username and password.

Solved by adding new handler for `email_confirmed` signal, "welcome" email message will be sent under following conditions a) email address was verified successfuly, b) they are staff, c) they  was never logged into app before, means they are "newly registered owners", who  does not know their login/password yet. 

Does it make any sense to you?

### Testing?

Yes, every feature/improvement was fully covered by tests.

### Anything Else?

Yes, you need to "write" welcome message before merging into main branch

Closes #1489 

### Review request
@tylerecouture

## Summary by CodeRabbit

- **New Features**
	- Simplified name fields in the `Profile` model by removing `first_name` and `last_name`, and added a `preferred_name` option.
	- Implemented email confirmation handling and sent welcome emails to new users.
	- Added a CSS rule to set `min-height: 31px` for the `.note-btn` class.

- **Tests**
	- Enhanced tests to cover setting of owner credentials and permission check for 'announcements:archived' endpoint.

### CODE REVIEW

**Comment by bashu** [comment]
to @tylerecouture 

> Once the owner user has verified the email (need to hook into that somehow?) for the first time, send an email with instructions for how to log in with the username and password.

TBD, will update this PR soon.

**Comment by tylerecouture** [comment]
@bashu is this still a WIP?

File: src/tenant/views.py [hunk 1]
```diff
@@ -76,6 +79,17 @@ def form_valid(self, form):
...
+            #
+            # ``generate_unique_username`` method returns a unique username from the combination of strings
+            # present in txts (first argument) iterable, for reference:
+            # https://docs.allauth.org/en/latest/account/advanced.html#creating-and-populating-user-instances
+            user_username(owner, get_adapter().generate_unique_username([
+                ".".join([owner.first_name, owner.last_name])]))
+
+            # set the owner's password to firstname-deckname-lastname
+            owner.set_password("-".join([owner.first_name, self.object.name, owner.last_name]).lower())
+
```
[the remark this answers is not in the document]
Comment (bashu) [reply src/tenant/views.py:92]: bad bot

**Comment by bashu** [comment]
to @tylerecouture where to put signal handler? 

there is one at [profile_manager/models.py](https://github.com/bytedeck/bytedeck/blob/develop/src/profile_manager/models.py#L468), not sure if it's a proper place to put it there. Anyway I'm going to put it random place, and you will suggest one later...

**Comment by bashu** [comment]
to @tylerecouture please review

**Comment by tylerecouture** [comment]
@bashu "b) they are staff" <-- instead of this, it should check if the user is the deck owner, not just staff, because this is not relevant to all staff, it is only relevant to the owner (as set in SiteConfig).

File: src/tenant/views.py [hunk 2]
```diff
@@ -76,6 +85,17 @@ def form_valid(self, form):
             owner.last_name = cleaned_data['last_name']
             owner.save()
 
+            # set the owner's username to firstname.lastname (instead of "owner")
+            #
+            # ``generate_unique_username`` method returns a unique username from the combination of strings
+            # present in txts (first argument) iterable, for reference:
+            # https://docs.allauth.org/en/latest/account/advanced.html#creating-and-populating-user-instances
+            user_username(owner, get_adapter().generate_unique_username([
```
Comment (tylerecouture) [inline src/tenant/views.py:93]: Is this necessary?  Seems over engineered. On a new deck there are no users yet, except "bytedeck_admin".  Why not just set it to f"{first_name}.{last_name}"?

File: src/tenant/views.py [hunk 3]
```diff
@@ -114,3 +133,30 @@ def post(self, request, *args, **kwargs):
...
+@receiver(email_confirmed, sender=EmailConfirmationHMAC)
+def email_confirmed_handler(email_address, **kwargs):
+    # Once the owner user has verified the email for the first time,
+    # send an email with instructions for how to log in with the username and password.
+    user = email_address.user
+
+    # just verified email for a first time and never been logged into app before
+    if user.last_login is not None:
+        return
+    if user.is_staff is not True:
```
Comment (tylerecouture) [inline src/tenant/views.py:147]: only the `siteconfig.owner`, not all staff.

File: src/tenant/views.py [hunk 4]
```diff
@@ -114,3 +133,30 @@ def post(self, request, *args, **kwargs):
...
+        return
+
+    # generate "welcome" email for new user
+    msg = get_template("tenant/email/welcome.txt").render(context={
+        "user": user,
+    })
+
+    # sending a text and HTML content combination
+    email = EmailMultiAlternatives(
+        f"Welcome to {SiteConfig.get().site_name_short}!",
```
Comment (tylerecouture) [inline src/tenant/views.py:157]: This should use a template too

**Review by tylerecouture — CHANGES_REQUESTED** [review]
Looking great.  Thanks for the thorough testing and good documentation. A couple small notes for you.

**Comment by tylerecouture** [comment]
@bashu this one is pretty much done, are you able to resolve the last few review comments?

File: src/tenant/views.py (hunk 2 above)
Comment (bashu) [reply src/tenant/views.py:93]: probably..., I will simplify it, if you are requesting

File: src/tenant/views.py (hunk 3 above)
Comment (bashu) [reply src/tenant/views.py:147]: done

File: src/tenant/views.py (hunk 4 above)
Comment (bashu) [reply src/tenant/views.py]: done

File: src/tenant/views.py [hunk 5]
```diff
@@ -114,3 +127,39 @@
...
+    })
+
+    # sending a text and HTML content combination
+    email = EmailMultiAlternatives(
+        subject,
+        body=textify(msg),  # convert msg to plain text, using textify utility
+        to=[user.email],
+    )
+    email.attach_alternative(msg, "text/html")
+    email.send()
```
[the remark this answers is not in the document]
Comment (bashu) [reply src/tenant/views.py]: nice try, but handled by Django

File: src/tenant/views.py [hunk 6]
```diff
@@ -76,6 +84,12 @@ def form_valid(self, form):
             owner.last_name = cleaned_data['last_name']
             owner.save()
 
+            # set the owner's username to firstname.lastname (instead of "owner")
+            user_username(owner, f"{owner.first_name}.{owner.last_name}")
+
+            # set the owner's password to firstname-deckname-lastname
+            owner.set_password("-".join([owner.first_name, self.object.name, owner.last_name]).lower())
+
```
[the remark this answers is not in the document]
Comment (bashu) [reply src/tenant/views.py:92]: good catch, but client requested simple password to be set by default

File: src/tenant/views.py [hunk 7]
```diff
@@ -76,6 +84,12 @@
             owner.last_name = cleaned_data['last_name']
             owner.save()
 
+            # set the owner's username to firstname.lastname (instead of "owner")
+            user_username(owner, f"{owner.first_name}.{owner.last_name}")
+
+            # set the owner's password to firstname-deckname-lastname
+            owner.set_password("-".join([owner.first_name, self.object.name, owner.last_name]).lower())
```
[the remark this answers is not in the document]
Comment (bashu) [reply src/tenant/views.py:91]: good catch, but client requested simple password to be set by default

File: src/tenant/views.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (bashu) [reply src/tenant/views.py:92]: won't work in this case

**Comment by bashu** [comment]
to @tylerecouture resolved, see new commits

**Comment by bashu** [comment]
to @tylerecouture and as usual tests fails, not related to this PR

**Review by tylerecouture — APPROVED** [review]

**Pull request merged**